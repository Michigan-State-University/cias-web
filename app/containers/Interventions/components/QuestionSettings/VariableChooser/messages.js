/*
 * VariableChooser Messages
 *
 * This contains all the text for the VariableChooser container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.VariableChooser';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Next Screen',
  },
});
