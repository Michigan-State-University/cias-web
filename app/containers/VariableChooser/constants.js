export const VIEWS = {
  VARIABLE: 'VARIABLE',
  SESSION: 'SESSION',
};
