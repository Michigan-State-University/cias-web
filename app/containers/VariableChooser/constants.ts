import { createContext } from 'react';

import { QuestionTypes, QuestionDTO } from 'models/Question';
import { SessionTypes } from 'models/Session';

export enum VIEWS {
  VARIABLE = 'VARIABLE',
  SESSION = 'SESSION',
  INTERVENTION = 'INTERVENTION',
}

interface ContextType {
  currentInterventionId: string;
  currentSessionId: string;
  currentView: string;
  includeAllSessions: boolean;
  includeAllVariables: boolean;
  includeCurrentQuestion: boolean;
  includeCurrentSession: boolean;
  includeNonDigitVariables: boolean;
  initialInterventionId: string;
  initialSessionId: string;
  isMultiIntervention: boolean;
  isMultiSession: boolean;
  organizationId: string;
  questionTypeWhitelist: QuestionTypes[];
  selectedQuestion: QuestionDTO | {};
  setCurrentView: (view: VIEWS) => void;
  sessionTypesWhiteList: SessionTypes[];
}

export const VariableChooserContext = createContext<ContextType>({
  currentInterventionId: '',
  currentSessionId: '',
  currentView: '',
  includeAllSessions: false,
  includeAllVariables: false,
  includeCurrentQuestion: false,
  includeCurrentSession: false,
  includeNonDigitVariables: false,
  initialInterventionId: '',
  initialSessionId: '',
  isMultiIntervention: false,
  isMultiSession: false,
  organizationId: '',
  questionTypeWhitelist: [],
  selectedQuestion: {},
  setCurrentView: (_) => _,
  sessionTypesWhiteList: [],
});
