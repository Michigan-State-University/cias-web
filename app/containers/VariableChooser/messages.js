/*
 * VariableChooser Messages
 *
 * This contains all the text for the VariableChooser container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.VariableChooser';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Next Screen',
  },
  noVariables: {
    id: `${scope}.noVariables`,
    defaultMessage: 'No variables defined yet',
  },
  noSessions: {
    id: `${scope}.noSessions`,
    defaultMessage: 'No sessions to choose',
  },
  selectedInterventionBadge: {
    id: `${scope}.selectedInterventionBadge`,
    defaultMessage: 'Current',
  },
});
