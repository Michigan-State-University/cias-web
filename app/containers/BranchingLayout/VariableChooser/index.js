import React, { useMemo, useRef } from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { colors, boxShadows } from 'theme';

import Column from 'components/Column';
import Row from 'components/Row';
import Box from 'components/Box';
import Text from 'components/Text';
import Img from 'components/Img';
import Badge from 'components/Badge';
import Loader from 'components/Loader';
import NoContent from 'components/NoContent';

import webpage from 'assets/svg/webpage-mouseover.svg';
import {
  makeSelectFilteredQuestions,
  makeSelectSelectedQuestion,
} from 'global/reducers/questions';

import Question from 'models/Session/Question';
import {
  getBranchingVariables,
  getEditVariables,
  getPreviousQuestions,
} from 'models/Session/utils';
import { nameQuestion, QuestionTypes } from 'models/Session/QuestionTypes';

import { htmlToPlainText } from 'utils/htmlToPlainText';
import useOutsideClick from 'utils/useOutsideClick';
import {
  makeSelectGetQuestionGroupLoader,
  makeSelectQuestionGroups,
} from 'global/reducers/questionGroups';
import messages from './messages';

const VariableChooser = ({
  intl: { formatMessage },
  onClick,
  questions,
  groups,
  selectedQuestion = {},
  visible,
  setOpen,
  loading,
  includeAllVariables,
  topPosition,
  includeCurrentQuestion,
  includeNonDigitVariables,
  questionTypeWhitelist,
  style,
}) => {
  const { id } = selectedQuestion;
  const filteredQuestions = useMemo(() => {
    const visibleQuestions = includeAllVariables
      ? questions
      : getPreviousQuestions(
          selectedQuestion,
          questions,
          groups,
          includeCurrentQuestion,
        );

    return visibleQuestions.filter(({ type }) =>
      questionTypeWhitelist.includes(type),
    );
  }, [selectedQuestion, questions, groups]);

  const variableGetter = includeNonDigitVariables
    ? getEditVariables
    : getBranchingVariables;

  const variables = variableGetter(filteredQuestions, {
    structure: 'flat',
    include: ['id', 'subtitle'],
  });

  const variableChooser = useRef(null);
  useOutsideClick(variableChooser, () => setOpen(false), visible);

  const displayContent = () => {
    const filteredVariables = variables?.length
      ? variables.filter(
          ({ variable }) => variable !== nameQuestion.reservedVariable,
        )
      : undefined;
    if (filteredVariables && filteredVariables.length)
      return filteredVariables.map((variable, index) => (
        <Row
          data-testid={`${id}-select-variable`}
          key={`${id}-select-variable-${index}`}
          mb={index !== variables.length - 1 && 15}
          onClick={() => onClick(variable.variable)}
          justify="between"
          align="center"
          clickable
        >
          <Row align="center">
            <Img src={webpage} mr={15} />
            <Text
              textOverflow="ellipsis"
              whiteSpace="pre"
              overflow="hidden"
              maxWidth={200}
              mr={20}
            >
              {htmlToPlainText(variable.subtitle)}
            </Text>
          </Row>
          <Badge
            data-cy="question-variable-chooser"
            maxWidth={300}
            color={colors.jungleGreen}
            bgWithOpacity
          >
            {variable.variable && variable.variable.trim()}
          </Badge>
        </Row>
      ));

    return (
      <Box padding={30}>
        <NoContent text={formatMessage(messages.noVariables)} />
      </Box>
    );
  };

  return (
    <Box
      ref={variableChooser}
      bg={colors.white}
      borderRadius={10}
      shadow={boxShadows.black}
      position="absolute"
      width="auto"
      right="0px"
      top={topPosition}
      {...(visible ? { zIndex: 1 } : { display: 'none' })}
      {...style}
    >
      <Row>
        <Box padding={8} filled>
          {loading && <Loader type="inline" />}
          {!loading && <Column>{displayContent()}</Column>}
        </Box>
      </Row>
    </Box>
  );
};

VariableChooser.propTypes = {
  intl: PropTypes.object,
  onClick: PropTypes.func.isRequired,
  questions: PropTypes.arrayOf(PropTypes.shape(Question)),
  groups: PropTypes.arrayOf(
    PropTypes.shape({ id: PropTypes.string, position: PropTypes.number }),
  ),
  selectedQuestion: PropTypes.shape(Question),
  visible: PropTypes.bool,
  loading: PropTypes.bool,
  includeAllVariables: PropTypes.bool,
  setOpen: PropTypes.func,
  topPosition: PropTypes.string,
  includeCurrentQuestion: PropTypes.bool,
  includeNonDigitVariables: PropTypes.bool,
  questionTypeWhitelist: PropTypes.arrayOf(PropTypes.string),
  style: PropTypes.object,
};

VariableChooser.defaultProps = {
  includeCurrentQuestion: true,
  questionTypeWhitelist: QuestionTypes.map(({ id }) => id),
};

const mapStateToProps = createStructuredSelector({
  questions: makeSelectFilteredQuestions(),
  groups: makeSelectQuestionGroups(),
  selectedQuestion: makeSelectSelectedQuestion(),
  loading: makeSelectGetQuestionGroupLoader(),
});

const withConnect = connect(mapStateToProps);

export default injectIntl(compose(withConnect)(VariableChooser));
