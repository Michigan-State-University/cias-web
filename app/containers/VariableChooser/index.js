import React, {
  memo,
  useCallback,
  useLayoutEffect,
  useMemo,
  useRef,
  useState,
} from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { injectReducer, injectSaga } from 'redux-injectors';

import {
  allCopyModalSagas,
  copyModalReducer,
} from 'global/reducers/copyModalReducer';

import { colors, boxShadows } from 'theme';

import { QuestionTypes } from 'models/Session/QuestionTypes';
import Question from 'models/Session/Question';

import useOutsideClick from 'utils/useOutsideClick';

import Row from 'components/Row';
import Box from 'components/Box';

import VariableView from './Components/VariableView';
import SessionView from './Components/SessionView';
import InterventionView from './Components/InterventionView';

import { VariableChooserContext, VIEWS } from './constants';

const VariableChooser = ({
  children,
  disabled,
  includeAllSessions,
  includeAllVariables,
  includeCurrentQuestion,
  includeCurrentSession,
  includeNonDigitVariables,
  interventionId: initialInterventionId,
  isMultiIntervention,
  isMultiSession,
  onClick,
  organizationId,
  placement,
  questionTypeWhitelist,
  selectedQuestion = {},
  sessionId: initialSessionId,
  topPosition,
  setIsOpen: propsSetOpen,
}) => {
  const variableChooser = useRef(null);

  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);
  const close = () => setIsOpen(false);

  useOutsideClick(variableChooser, close, isOpen);

  const initialView = useMemo(() => {
    if (isMultiIntervention) return VIEWS.INTERVENTION;

    if (isMultiSession && !includeCurrentSession) return VIEWS.SESSION;

    return VIEWS.VARIABLE;
  }, [isMultiIntervention, isMultiSession, includeCurrentSession]);

  const [currentView, setCurrentView] = useState(initialView);
  const toSessionView = () => setCurrentView(VIEWS.SESSION);
  const toVariableView = () => setCurrentView(VIEWS.VARIABLE);

  const [currentInterventionId, setCurrentInterventionId] = useState(
    initialInterventionId,
  );
  const resetCurrentInterventionId = () =>
    currentInterventionId !== initialInterventionId &&
    setCurrentInterventionId(initialInterventionId);

  const [currentSessionId, setCurrentSessionId] = useState(initialSessionId);
  const resetCurrentSessionId = () =>
    currentSessionId !== initialSessionId &&
    setCurrentSessionId(initialSessionId);

  const resetVariableChooser = () => {
    resetCurrentSessionId();
    resetCurrentInterventionId();
    setCurrentView(initialView);
  };

  useLayoutEffect(() => {
    if (isOpen) resetVariableChooser();
    if (propsSetOpen) propsSetOpen(isOpen);
  }, [isOpen]);

  const handleOnVariableClick = useCallback(
    variable => {
      onClick(variable);
      close();
    },
    [onClick],
  );

  const handleOnSessionClick = useCallback(sessionId => {
    setCurrentSessionId(sessionId);
    toVariableView();
  }, []);

  const handleOnInterventionClick = useCallback(interventionId => {
    setCurrentInterventionId(interventionId);
    toSessionView();
  }, []);

  const displayContent = useCallback(() => {
    if (!isOpen) return null;

    switch (currentView) {
      case VIEWS.VARIABLE:
        return <VariableView onClick={handleOnVariableClick} />;

      case VIEWS.SESSION:
        return <SessionView onClick={handleOnSessionClick} />;

      case VIEWS.INTERVENTION:
        return <InterventionView onClick={handleOnInterventionClick} />;

      default:
        return null;
    }
  }, [
    currentView,
    handleOnInterventionClick,
    handleOnSessionClick,
    handleOnVariableClick,
    isOpen,
  ]);

  return (
    <Box position="relative">
      <Box onClick={toggle} disabled={disabled} clickable>
        {children}
      </Box>
      <Box
        ref={variableChooser}
        bg={colors.white}
        borderRadius={10}
        shadow={boxShadows.black}
        position="absolute"
        width="auto"
        top={topPosition}
        {...(isOpen ? { zIndex: 1, [placement]: '0px' } : { display: 'none' })}
      >
        <Row>
          <Box padding={8} filled>
            <VariableChooserContext.Provider
              value={{
                currentInterventionId,
                currentSessionId,
                currentView,
                includeAllSessions,
                includeAllVariables,
                includeCurrentQuestion,
                includeCurrentSession,
                includeNonDigitVariables,
                initialInterventionId,
                initialSessionId,
                isMultiIntervention,
                isMultiSession,
                organizationId,
                questionTypeWhitelist,
                selectedQuestion,
                setCurrentView,
              }}
            >
              {displayContent()}
            </VariableChooserContext.Provider>
          </Box>
        </Row>
      </Box>
    </Box>
  );
};

VariableChooser.propTypes = {
  children: PropTypes.node,
  disabled: PropTypes.bool,
  includeAllSessions: PropTypes.bool,
  includeAllVariables: PropTypes.bool,
  includeCurrentQuestion: PropTypes.bool,
  includeCurrentSession: PropTypes.bool,
  includeNonDigitVariables: PropTypes.bool,
  interventionId: PropTypes.string,
  isMultiIntervention: PropTypes.bool,
  isMultiSession: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  organizationId: PropTypes.string,
  placement: PropTypes.oneOf(['left', 'right']),
  questionTypeWhitelist: PropTypes.arrayOf(PropTypes.string),
  questions: PropTypes.arrayOf(PropTypes.shape(Question)),
  selectedQuestion: PropTypes.shape(Question),
  sessionId: PropTypes.string,
  topPosition: PropTypes.string,
  setIsOpen: PropTypes.func,
};

VariableChooser.defaultProps = {
  includeCurrentQuestion: true,
  includeCurrentSession: true,
  placement: 'right',
  questionTypeWhitelist: QuestionTypes.map(({ id }) => id),
};

const reduxInjectors = [
  injectReducer({ key: 'copyModal', reducer: copyModalReducer }),
  injectSaga({ key: 'copyModal', saga: allCopyModalSagas }),
];

export default compose(
  ...reduxInjectors,
  memo,
)(VariableChooser);
