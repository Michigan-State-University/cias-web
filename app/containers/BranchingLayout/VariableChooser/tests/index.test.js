import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { Provider } from 'react-redux';

import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'i18n';
import { createStore } from 'redux';
import VariableChooser from '../index';

describe('<VariableChooser />', () => {
  let store;
  const reducer = state => state;
  const initialState = {
    questions: {
      questions: [],
    },
  };

  const props = {
    onClick: jest.fn(),
  };

  beforeAll(() => {
    store = createStore(reducer, initialState);
    store.runSaga = () => {};
    store.injectedReducers = {};
    store.injectedSagas = {};
  });

  it('should match the snapshot without variables', () => {
    const renderedComponent = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <VariableChooser {...props} />
        </IntlProvider>
      </Provider>,
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it('should contain <NoContent /> when without variables', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <VariableChooser {...props} />
        </IntlProvider>
      </Provider>,
    );

    const noContentComponent = getByTestId('no-content');

    expect(noContentComponent).not.toEqual(null);
  });

  it('should not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');

    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <VariableChooser {...props} />
        </IntlProvider>
      </Provider>,
    );

    expect(spy).not.toHaveBeenCalled();
  });

  it('should match the snapshot with variables', () => {
    store = createStore(reducer, {
      questions: {
        questions: [mockSingleQuestion()],
      },
    });

    const renderedComponent = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <VariableChooser {...props} />
        </IntlProvider>
      </Provider>,
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render <NoContent /> when all variables are empty strings', () => {
    store = createStore(reducer, {
      questions: {
        questions: [mockSingleQuestion(1, false)],
      },
    });

    const { getByTestId } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <VariableChooser {...props} />
        </IntlProvider>
      </Provider>,
    );

    const noContentComponent = getByTestId('no-content');

    expect(noContentComponent).not.toEqual(null);
  });

  it('should render list of variables', () => {
    const question = mockSingleQuestion(1, true);

    store = createStore(reducer, {
      questions: {
        questions: [question],
        selectedQuestion: question.id,
      },
    });

    const { getAllByTestId } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <VariableChooser {...props} />
        </IntlProvider>
      </Provider>,
    );

    const variableComponentList = getAllByTestId(
      `${question.id}-select-variable`,
    );

    expect(variableComponentList).toHaveLength(1);
  });

  it('should invoke onClick', () => {
    const question = mockSingleQuestion(1, true);

    store = createStore(reducer, {
      questions: {
        questions: [question],
        selectedQuestion: question.id,
      },
    });

    const { getByText } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <VariableChooser {...props} />
        </IntlProvider>
      </Provider>,
    );
    const row = getByText(`Test subtitle 1`);
    fireEvent.click(row);
    expect(props.onClick).toHaveBeenCalledWith('var_single_1');
  });
});

const mockSingleQuestion = (suffix = 1, hasVariable = true) => ({
  id: `test-id-${suffix}`,
  title: `Test title ${suffix}`,
  subtitle: `Test subtitle ${suffix}`,
  type: 'Question::Single',
  body: {
    variable: { name: hasVariable ? `var_single_${suffix}` : '' },
  },
});
