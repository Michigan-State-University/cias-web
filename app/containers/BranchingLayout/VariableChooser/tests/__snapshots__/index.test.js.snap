// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VariableChooser /> should match the snapshot with variables 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  justify-items: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  width: auto;
  position: absolute;
  display: none;
  background-color: rgba(255,255,255,1);
  border-radius: 10px;
  box-shadow: 0px 0px 50px rgba(0,0,0,0.08);
  border-radius: 10px;
  right: 0px;
}

.c2 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  padding: 8px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c7 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  margin-right: 20px;
  color: #2F3850;
  white-space: pre;
  text-overflow: ellipsis;
  max-width: 200px;
  overflow: hidden;
}

.c6 {
  width: auto;
  height: auto;
  margin-right: 15px;
}

.c8 {
  min-height: 24px;
  min-width: 25px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  padding: 5px 10px;
  font-size: 13px;
  font-weight: 600;
  border-radius: 4px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  background-color: rgba(16,121,105,0.1);
  color: rgba(16,121,105,1);
  max-width: 300px;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div>
      <div
        class="c0"
        display="none"
        width="auto"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div
                class="c4"
                data-testid="undefined-select-variable"
              >
                <div
                  class="c5"
                >
                  <img
                    class="c6"
                    src="IMAGE_MOCK"
                  />
                  <p
                    class="c7"
                    overflow="hidden"
                  >
                    Test subtitle 1
                  </p>
                </div>
                <div
                  class="c8"
                  color="#107969"
                  data-cy="question-variable-chooser"
                >
                  var_single_1
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-1vhutan-0 bBMNIf"
      display="none"
      width="auto"
    >
      <div
        class="Row-y1cs0u-0 dskFsL"
      >
        <div
          class="Box-sc-1vhutan-0 buKpDj"
        >
          <div
            class="Column-sc-4zswzq-0 ekpwZH"
          >
            <div
              class="Row-y1cs0u-0 dLpgLs"
              data-testid="undefined-select-variable"
            >
              <div
                class="Row-y1cs0u-0 ecllgQ"
              >
                <img
                  class="Img-sc-55j3su-0 gWNtSX"
                  src="IMAGE_MOCK"
                />
                <p
                  class="Text-e4p0w3-0 cHolUu"
                  overflow="hidden"
                >
                  Test subtitle 1
                </p>
              </div>
              <div
                class="Badge-n0b6mf-0 AXzkR"
                color="#107969"
                data-cy="question-variable-chooser"
              >
                var_single_1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<VariableChooser /> should match the snapshot without variables 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
}

.c0 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  width: auto;
  position: absolute;
  display: none;
  background-color: rgba(255,255,255,1);
  border-radius: 10px;
  box-shadow: 0px 0px 50px rgba(0,0,0,0.08);
  border-radius: 10px;
  right: 0px;
}

.c2 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  padding: 8px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  padding: 30px;
}

.c5 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  width: 100%;
  height: 100%;
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 0;
  border-radius: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  width: auto;
  height: auto;
}

.c8 {
  margin: 0;
  font-family: DM Sans;
  font-size: 18px;
  line-height: 130%;
  margin-top: 15px;
  color: rgba(176,186,201,1);
  color: rgba(176,186,201,1);
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<body>
    <div>
      <div
        class="c0"
        display="none"
        width="auto"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div
                class="c4"
              >
                <div
                  class="c5"
                  data-testid="no-content"
                  display="flex"
                  height="100%"
                  width="100%"
                >
                  <div
                    class="c6"
                  >
                    <img
                      alt="No content"
                      class="c7"
                      src="IMAGE_MOCK"
                    />
                    <h2
                      class="c8"
                      color="#B0BAC9"
                    >
                      No variables defined yet
                    </h2>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Box-sc-1vhutan-0 bBMNIf"
      display="none"
      width="auto"
    >
      <div
        class="Row-y1cs0u-0 dskFsL"
      >
        <div
          class="Box-sc-1vhutan-0 buKpDj"
        >
          <div
            class="Column-sc-4zswzq-0 ekpwZH"
          >
            <div
              class="Box-sc-1vhutan-0 dbODYP"
            >
              <div
                class="Box-sc-1vhutan-0 kTgzBZ"
                data-testid="no-content"
                display="flex"
                height="100%"
                width="100%"
              >
                <div
                  class="Column-sc-4zswzq-0 eExeGs"
                >
                  <img
                    alt="No content"
                    class="Img-sc-55j3su-0 cDhXoN"
                    src="IMAGE_MOCK"
                  />
                  <h2
                    class="H2-uw8lbl-0 bnQANz"
                    color="#B0BAC9"
                  >
                    No variables defined yet
                  </h2>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
